`Question: 1` A company wants to estimate the average time spent by customers on their website per session. They collect a random sample of 100 customers and find that the sample mean time spent is 4.5 minutes with a sample standard deviation of 1.2 minutes. Calculate a 99% confidence interval for the true population mean time spent on the website per session.


`Question 2:` A car manufacturer is interested in estimating the mean gas mileage of their new SUV. A sample of 25 SUVs is taken, and their mean gas mileage is found to be 28.6 miles per gallon with a standard deviation of 2.8 miles per gallon. Calculate a 95% confidence interval for the true mean gas mileage of the SUV.

`Question -3:` Dataset - India temperature dataset.

Data link: https://drive.google.com/file/d/1whb8Rjdk9TBKOwVi8NwZD1Tc8Tdv7iEc/view?usp=share_link

Kaggle Link : (Just take IND data for this question, have to apply filtering). https://www.kaggle.com/datasets/berkeleyearth/climate-change-earth-surface-temperature-data

A climate research organization wants to estimate the average temperature of a certain country. They collect temperature data for 2613 days but due to certain limitations, they only have information about the average temperature for 2508 days. The organization assumes that the population follows a normal distribution and wants to estimate the population mean temperature with a 95% confidence interval.

* Standard deviation is assumed as given data standard deviation. z Procedure
* Standard deviation is not given. Apply t Procedure

`Question 5:` BMW Dataset : https://drive.google.com/file/d/18amVv1WKKHEmwyvVy0RuXp4A6xrGOelc/view?usp=share_link



* Task 1:
The sales manager of a used car company wants to know what is the average selling price of all the used bmw cars.
The analyst can collect only a sample of sales of 500 cars in the area. Since this estimate is going to be used by the company to strategize sales of his company, the sample mean should be a good approximation of all the account.
What level of confidence is the sales manager going to be satisfied with? What +/- interval number is going to be acceptable?

* Task 2:
In addition to the price of the car, the manager also wants to now know the average mileage that the car has been driven. But the manager does not have the population standard deviation for the mileage. But the mileage data is available only from the 25 cars that they have sold so far.
How do the analyst approach this problem to calculate 95% confidence interval, with only 25 samples?

* Task 3:
The manager is not happy with both the intervals (intervals from task 1 and 2) as the interval for the confidence is very high. The manager now asks the analyst to estimate the average price of the car (similar to task 1) but within a bound of 750 from the mean with 95% condidence level. How many sample does the analyst have to collect to arrive at this confidence interval level?

* Task 4:
Conversely, after the analyst started collecting the data, after a week he was only able to collect for 420 samples, though he needs 540 samples for Bound of 1,000, if he has to do the analysis now, what is the best interval he can acheive for 95% confidence levels?

In [3]:
import pandas as pd 
import seaborn as sns 
import numpy as np
import scipy
df = pd.read_csv('bmw.csv')

In [10]:
df.head()

model  year  price transmission  mileage fuelType  tax   mpg  \
0   5 Series  2014  11200    Automatic    67068   Diesel  125  57.6   
1   6 Series  2018  27000    Automatic    14827   Petrol  145  42.8   
2   5 Series  2016  16000    Automatic    62794   Diesel  160  51.4   
3   1 Series  2017  12750    Automatic    26676   Diesel  145  72.4   
4   7 Series  2014  14500    Automatic    39554   Diesel  160  50.4   

   engineSize  
0         2.0  
1         2.0  
2         3.0  
3         1.5  
4         3.0

# Task-1

In [11]:
n = 500

In [22]:
sample = df['price'].sample(n, random_state = 42)
sample = np.array(sample)

In [23]:
sample_mean, sample_std = np.mean(sample), np.std(sample, ddof = 1)

In [24]:
print(sample_mean, sample_std)

23011.168 10979.65860735406


In [25]:
confidence_level = 0.95

In [26]:
z_critical = scipy.stats.norm.ppf(1-(1-confidence_level)/2)

In [91]:
point_estimate = sample_mean 
margin_of_error = z_critical*sample_std/np.sqrt(n)
print(margin_of_error)

4303.947086591833


In [28]:
CI = [point_estimate-margin_of_error, point_estimate+margin_of_error]

In [29]:
CI

[22048.77617428185, 23973.559825718152]

# Task-2

In [67]:
n=25
confidence_level = 0.95 

In [85]:
data = df['mileage'].sample(25, random_state = 1)
data = np.array(data)

In [86]:
data_mean, data_std = np.mean(data),np.std(data,ddof=1)
print(f"mean of data : {data_mean} and standard deviation of data : {data_std}")

mean of data : 25148.68 and standard deviation of data : 17421.996206233085


In [87]:
t_critical = scipy.stats.t.ppf(1-(1-confidence_level)/2, df=n-1)

In [88]:
margin_error = t_critical*data_std/np.sqrt(n)
print(margin_error)

7191.446582146658


In [89]:
CI = [data_mean-margin_error,data_mean+margin_error]
CI

[17957.233417853342, 32340.12658214666]

# Task-3

In [97]:
std = df['price'].std()
B=750 

In [98]:
cl = 0.95
z_critical = scipy.stats.norm.ppf(0.975)

In [101]:
n = (z_critical**2)*(std**2)/B**2

In [106]:
n = round(n)
n

890

### approximately 890 samples are requied to get margin of error 750

In [155]:
data = df['price'].sample(n,random_state = 1)
data = np.array(data)

In [156]:
data_mean,data_std = data.mean(), data.std()

In [157]:
margin_of_error = z_critical*data_std/np.sqrt(n)

In [158]:
margin_of_error

750.4659556956817

# Task-4

In [161]:
# in question we need to find margin of error at n=420 

In [164]:
n=420
data = df['price'].sample(n)
data = np.array(data)
data_std = data.std()
margin_of_error = z_critical*data_std/np.sqrt(420)

In [165]:
margin_of_error

1104.2609430550476